pkg/exemplars/proxy.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package exemplars
+
+import (
+	"context"
+	"io"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/exemplars/exemplarspb"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Proxy implements exemplarspb.Exemplars gRPC that fanouts requests to
+// given exemplarspb.Exemplars and de-duplication on the way.
```

Comment:
Let's update this comment. Currently, Proxy itself does not deduplicate. Instead, the it seems to be a responsibility of the caller of proxy.Exemplars to deduplication. Alternatively, we can make it so that the Proxy needs to deduplication.